tests/codex/sales/testreservations.nim

```diff
@@ -148,6 +152,39 @@ asyncchecksuite "Reservations module":
     check created.isErr
     check created.error of BytesOutOfBoundsError
 
+  test "cannot create reservation larger than availability size - concurrency test":
+    proc concurrencyTest(): Future[void] {.async.} =
+      let availability = createAvailability()
+      let one = reservations.createReservation(
+        availability.id,
+        availability.totalSize - 1,
+        RequestId.example,
+        UInt256.example
+      )
+
+      let two = reservations.createReservation(
+        availability.id,
+        availability.totalSize,
+        RequestId.example,
+        UInt256.example
+      )
+
+      let oneResult = await one
+      let twoResult = await two
+
+      check oneResult.isErr or twoResult.isErr
+      if oneResult.isErr:
+        check oneResult.error of BytesOutOfBoundsError
+      if twoResult.isErr:
+        check twoResult.error of BytesOutOfBoundsError
```

Comment:
i think chronos has a better primitive for this: `allFinished`

Not saying you should do this, but was writing out a different idea and thought it would be worth mentioning another way to achieve the same:
```nim
import std/sugar

let results = allFinished(one, two)
check results.any(r => r.error of BytesOutOfBoundsError)
check results.all(r => r.isOk or r.error of BytesOutOfBoundsError)
```

Reply:
Ohhh right, I was trying make it work with `allFinishedThrowing` and `allFinishedResult`, but it had its problems and kind of forgot the simplest of these `allFinished` 😅 Thanks! Will update...

Reply:
Well, FYI this does not work. Mainly because `allFinished` will return sequence of futures and it does not "unwrap" the underlying result as you (and I was as well) expected. I have again tried to make this work for some time without success so I will just leave it as is.

Reply:
you need to use `.value` i think to get the underlying value, eg 

```nim
import std/sugar

let results = allFinished(one, two)
check results.any(r => r.value.error of BytesOutOfBoundsError)
check results.all(r => r.value.isOk or r.value.error of BytesOutOfBoundsError)
```